### ISSUE TITLE
Option for compact headers

### ISSUE THREAD

**Issue by davestewart** [haltcase/tablemark#16]
Hello!

Nice little lib.

Unfortunately, WebStorm complains about the formatting of the headers:

```md
| first name | how old | are they cool |
| :--------- | :-----: | :------------ |
```

It seems it would prefer:

```md
| first name | how old | are they cool |
|:-----------|:-------:|:--------------|
```

Could we add an option for that?

Looking at the code, a post-generation regex around [line 49](https://github.com/haltcase/tablemark/blob/master/src/index.ts#L49) would probably be simplest option:

```js
separator.replace(/ (:?\-+:?) /g, (all, inner) => inner.replace('-', '---'))
```

I'm happy to PR this.

**Comment by haltcase** [comment]
Hey @davestewart, thanks for pointing this out. I haven't used WebStorm myself for a handful of years at this point. Could you provide more info about how it complains? I'd like to understand what sort of issue it presents before we make changes, e.g., is it an error message, lint warning, etc.

If we do make changes, I don't super love the idea of a post-generation text replacement and would rather we implement it during in the "rendering" phase itself somehow, but we'll get to that later (and I'd be happy to accept a PR once we've hit a good solution).

**Comment by davestewart** [comment]
Hey Bo,

Thanks for the prompt reply!

Sure, this is what it doesn't like, and what it reformats to:

![image](https://github.com/haltcase/tablemark/assets/132681/57dfd68e-0c25-4184-9d21-9977580f2284)

I hit `Opt+Enter` then choose to reformat, and it removes the spaces:

![image](https://github.com/haltcase/tablemark/assets/132681/f5889176-7d81-4e2c-a8cc-ff349ffcfd3b)

I totally understand you wanting to do it "the correct way"; perhaps the `line()` function could be adjusted to take an optional padding parameter?

Right now the regex works in my code (I process the whole table) but totally get you want to make this more robust!

**Comment by haltcase** [comment]
Released in [v3.1.0](https://github.com/haltcase/tablemark/releases/tag/v3.1.0)
